### PR TITLE
pipewire: 0.3.23 -> 0.3.24

### PR DESCRIPTION
**Pull request by jansol**
You know the drill. Seems to work out of the box with default configs. Not sure how nice changing config is atm (especially with the new arrays for `context.modules` and possibly other places). I'll try to update this PR based on feedback if I have time before it gets merged. If not, I'll do a separate PR or add the changes when the next release drops.

###### Motivation for this change

https://gitlab.freedesktop.org/pipewire/pipewire/-/releases/0.3.24

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/desktops/pipewire/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Updating
+
+1. Update the version & hash in pkgs/development/libraries/pipewire/default.nix
+2. run `nix build -I nixpkgs=/path/to/nixpkgs/checkout` nixpkgs.pipewire nixpkgs.pipewire.mediaSession
```
Comment (jtojnar) [inline nixos/modules/services/desktops/pipewire/README.md:4]:
```suggestion
2. run `nix build -f /path/to/nixpkgs/checkout pipewire pipewire.mediaSession`
```

Comment (jansol) [reply nixos/modules/services/desktops/pipewire/README.md:4]: ty, that was the command that I'd been using but this one seems more ergonomic

**Comment by ashkitten** [comment]
works for me

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116811` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>libsForQt512.xdg-desktop-portal-kde</li>
<li>libsForQt514.xdg-desktop-portal-kde</li>
</ul>
</details>
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
29 packages built:
</summary>
<ul>
<li>adapta-gtk-theme</li>
<li>adoptopenjdk-icedtea-web</li>
<li>chrome-gnome-shell</li>
<li>chromium</li>
<li>firefox-esr-78-unwrapped (firefoxPackages.firefox-esr-78)</li>
<li>xulrunner (firefox-unwrapped)</li>
<li>gnome3.gnome-control-center (gnome3.gnome_control_center)</li>
<li>gnome3.gnome-remote-desktop</li>
<li>gnome3.gnome-session (gnome3.gnome_session)</li>
<li>gnome3.gnome-shell (gnome3.gnome_shell)</li>
<li>gnome3.gnome-terminal (gnome3.gnome_terminal)</li>
<li>gnome3.gnome-tweaks (gnome3.gnome-tweak-tool)</li>
<li>gnome3.gpaste</li>
<li>gnome3.mutter</li>
<li>gnome3.pomodoro</li>
<li>gnomeExtensions.easyScreenCast</li>
<li>gnomeExtensions.gsconnect</li>
<li>gnomeExtensions.night-theme-switcher</li>
<li>gnomeExtensions.system-monitor</li>
<li>libsForQt5.xdg-desktop-portal-kde (libsForQt515.xdg-desktop-portal-kde ,plasma5Packages.xdg-desktop-portal-kde)</li>
<li>pantheon.elementary-session-settings</li>
<li>pipewire</li>
<li>pulseeffects-pw</li>
<li>puppeteer-cli</li>
<li>slimerjs</li>
<li>ungoogled-chromium</li>
<li>xdg-desktop-portal</li>
<li>xdg-desktop-portal-gtk</li>
<li>xdg-desktop-portal-wlr</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
adapta-gtk-theme:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/data/themes/adapta/default.nix:45:5:

```
   |
45 |     license = with licenses; [ gpl2 cc-by-sa-30 ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Pull request merged**